src/components/UIElements/ButtonWithActions.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+	<div
+		:class="`inline-block relative font-bold text-white shadow-md rounded hover:rounded focus:outline-none bg-${color}-500`">
+		<button
+			:class="`px-4 hover:bg-${color}-600`"
+			type="button"
+			@click.prevent="() => {actions[defaultActionIndex].action(); showDropdownButtons=false;}"
+		>
+			{{ actions[defaultActionIndex].label }}
+		</button>
+		<button
+			v-tooltip.right="userSettings.showTooltips
+						? 'Save & ...'
+						: { visible: false }
+			"
+			:class="`hover:bg-${color}-600 hover:rounded-r`"
+			type="button"
+			@click.prevent="showDropdownButtons = !showDropdownButtons"
+		>
+			<span v-if="!showDropdownButtons" class="material-icons">keyboard_arrow_{{ direction }}</span>
+			<span v-else class="material-icons">keyboard_arrow_{{ direction === 'up' ? 'down' : 'up' }}</span>
+		</button>
+		<div :class="classesForDirection[direction]" class="" :style="{top: `${dropdownOffset}px`}" v-if="showDropdownButtons" ref="dropdown">
+			<template
+				v-for="(action, i) in actions"
+			>
+				<button
+					:class="`w-full px-4 hover:bg-${color}-600`"
+					type="button"
+					v-if="i !== defaultActionIndex"
+					@click.prevent="() => {action.action(); showDropdownButtons=false;}"
+				>
+					{{ action.label }}
+				</button>
+			</template>
+		</div>
+	</div>
+</template>
+
+<script>
+	export default {
+		name: 'ButtonWithActions',
+		props: {
+			defaultActionIndex: {
+				required: false,
+				type: Number,
+				default: 0
+			},
+			direction: {
+				required: false,
+				type: String,
+				default: 'down'
+			},
+			actions: {
+				required: true,
+				type: Array
+			},
+			color: {
+				required: false,
+				type: String,
+				default: 'green'
+			},
+		},
+		computed: {
+			userSettings() {
+				return this.$store.getters.getUserSettings ?? {};
+			},
+		},
+		watch: {
+			showDropdownButtons() {
+				this.$nextTick(() => {
+					this.dropdownOffset = this.direction === 'up' ? -this.$refs.dropdown.clientHeight : 0;
+				})
+			}
+		},
+		data () {
+			return {
+				showDropdownButtons: false,
+				dropdownOffset: 0,
+				classesForDirection: {
+					down: `absolute z-50 w-full bg-${this.color}-700`,
+					up: `absolute z-50 w-full bg-${this.color}-700`
+				}
```

Comment:
don't pass such props that will change TW classes. Just pass TW class instead

Reply:
I fixed that